### PR TITLE
Add contributing doc

### PR DESCRIPTION
**Pull request by rmccue**
This lays out the philosophy behind the project, and how it should be maintained.

In addition to this, I'm adding more committers to the project: @dd32 and @ozh have both been contributing for a while; @robwilde and @ptasker both volunteered ([here](https://twitter.com/mrwilde/status/870627445589762048) and [here](https://twitter.com/petetasker/status/870631971986210816)) via Twitter to help out.

If any clarification is needed on these contribution notes, let me know; otherwise, any of you are welcome to review this PR, and merge it if accepted :)

### CODE REVIEW

**Comment by ptasker** [comment]
Contributing doc looks good but looks like Travis is failing, preventing merge, probably because of #279.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+7. PR will be merged.
+
+Notes:
+
+* All code needs to go through peer review. Committers may not merge their own PR.
+* PRs should **never be squashed or rebased**. This includes when merging. Keeping the history is important for tracking motivation behind changes later.
+
+## Compatibility
+
+All code in Requests must be compatible with PHP 5.2. Requests is used in WordPress, and hence is tied to its compatibility.
```
Comment (TysonAndre) [inline CONTRIBUTING.md]: Link this text to https://wordpress.org/about/requirements/ (minimum supported version), so we know if it ever goes out of date?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Requests is also bound to not break backwards compatibility. In semantic versioning terms, this means we will be in the 1.x release cycle for the forseeable future.
+
+Generally speaking, the goal of Requests is to provide an interface that smoothes the differences across various server setups. That means there shouldn't be any code relying on any extensions (even if they're common) unless we also provide a pure-PHP version (that is, we can use extensions to improve performance when available). Only two extensions are currently used: SPL, and cURL. SPL is always available from PHP 5.3 onwards, and is rarely disabled in PHP 5.2, so this is an acceptable exception. cURL is only used for better performance over sockets.
```
Comment (TysonAndre) [inline CONTRIBUTING.md]: nit: split up sentences into multiple lines, should still be rendered as a single line in HTML, and version history is easier to track

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+3. File a PR. If it isn't ready for merge yet, note that in the description. If your PR closes an existing issue, add "fixes #xxx" to the message, so that the issue will be closed when the PR is merged.
+4. If needed, iterate on the code until it is ready. This includes adding unit tests. When you're ready, comment that the PR is complete.
+5. A committer will review your code and offer you feedback.
+6. Update with the feedback as necessary.
+7. PR will be merged.
+
+Notes:
+
+* All code needs to go through peer review. Committers may not merge their own PR.
+* PRs should **never be squashed or rebased**. This includes when merging. Keeping the history is important for tracking motivation behind changes later.
```
Comment (TysonAndre) [inline CONTRIBUTING.md]: `**, even when merging` (reword sentence fragment?)

File: CONTRIBUTING.md (hunk 2 above)
Comment (rmccue) [inline CONTRIBUTING.md]: Good thinking; I used to wrap to 80 cols, which caused reflows, so gave up and just did the whole para on one line, but per-sentence is way better.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+3. File a PR. If it isn't ready for merge yet, note that in the description. If your PR closes an existing issue, add "fixes #xxx" to the message, so that the issue will be closed when the PR is merged.
+4. If needed, iterate on the code until it is ready. This includes adding unit tests. When you're ready, comment that the PR is complete.
+5. A committer will review your code and offer you feedback.
+6. Update with the feedback as necessary.
+7. PR will be merged.
+
+Notes:
+
+* All code needs to go through peer review. Committers may not merge their own PR.
+* PRs should **never be squashed or rebased**, even when merging. Keeping the history is important for tracking motivation behind changes later.
```
Comment (jrfnl) [inline CONTRIBUTING.md]: I'd like to suggest preferring rebases over back-merges. It still keeps the PR history intact and makes for a much cleaner repo history.

Also see: https://blog.developer.atlassian.com/stop-foxtrots-now/

**Comment by jrfnl** [comment]
@rmccue Would you mind rebasing/back-merging the PR so it becomes mergable ?

**Comment by jrfnl** [comment]
Just wondering: should the unit test section contain some information about running the unit tests locally with the proxy scripts ?

**Pull request closed without merging**